.github/workflows/node.js.yml

```diff
@@ -33,6 +33,12 @@ jobs:
                   cache: 'npm'
             - run: npm ci
             - run: npm run build --if-present
+            - run: npm run start &
```

Comment:
ChatGPT suggests following when asked about "Do you have example config for github actions to run selenium tests with TypeScript react app ?":

Would it work ? I don't like the while loop below

```
name: Selenium Tests

on:
  push:
    branches:
      - main

jobs:
  build-and-test:
    runs-on: ubuntu-latest

    steps:
      - name: Checkout code
        uses: actions/checkout@v2

      - name: Set up Node.js
        uses: actions/setup-node@v2
        with:
          node-version: 14

      - name: Install dependencies
        run: npm install

      - name: Build app
        run: npm run build

      - name: Run Selenium tests
        env:
          HEADLESS: true  # Optional: Set to 'false' if you want to see the browser GUI during tests
        run: |
          npm install webdriverio@7 selenium-standalone@latest
          npx selenium-standalone install
          npx selenium-standalone start &
          npm test
```

Reply:
No, I'm pretty sure it won't work. Selenium standalone is used to start a Selenium server, which allows for testing remotely and in parallel on multiple machines. The app still has to be running somewhere for tests to work (Webdriver connects to the app on this line: `await driver.get('http://localhost:3000');`). Therefore, I think there is no way to avoid launching the app in the background and then waiting for it to start. I can replace the while loop with a fixed time sleep command instead.

Reply:
What about something without fixed waiting time, like https://www.npmjs.com/package/wait-on

Following ons-not-working-after-npm-start
```
npm start & npx wait-on http://localhost:3000
```

Reply:
I got it to work using github actions wrapper for the wait-on package: https://github.com/marketplace/actions/wait-on. I tried to do it like this: `npm install wait-on && npx wait-on http://localhost:3000` - it worked for node 16, but on node 18 it waited forever and timed out after 360 minutes